ingestion_server/ingestion_server/api.py

```diff
@@ -47,124 +47,170 @@ def _get_base_url(req):
         parsed = urlparse(req.url)
         return parsed.scheme + "://" + parsed.netloc
 
-    @staticmethod
-    def _validate_create_task(request):
+    @jsonschema.validate(
+        req_schema={
+            "type": "object",
+            "properties": {
+                "model": {"type": "string", "enum": MEDIA_TYPES},
+                "action": {
+                    "type": "string",
+                    "enum": list(task_type.name for task_type in TaskTypes),
+                },
+                # Accepts all forms described in the PostgreSQL documentation:
+                # https://www.postgresql.org/docs/current/datatype-datetime.html
+                "since_date": {"type": "string"},
+                "index_suffix": {"type": "string"},
+                "alias": {"type": "string"},
+            },
+            "required": ["model", "action"],
+            "allOf": [
+                {
+                    "if": {
+                        "properties": {"action": {"const": TaskTypes.UPDATE_INDEX.name}}
+                    },
+                    "then": {"required": ["index_suffix", "since_date"]},
+                },
+                {
+                    "if": {
+                        "properties": {"action": {"const": TaskTypes.POINT_ALIAS.name}}
+                    },
+                    "then": {"required": ["index_suffix", "alias"]},
+                },
+            ],
+        }
+    )
+    def on_post(self, req, res):
```

Comment:
What happens if `req` is not validated?
I don't quite understand how this validation works and where `get_media` method is in `req`, if it's not in the schema...

Reply:
Falcon provides a built-in decorator `jsonschema.validate` to validate the body of the incoming request. It will run the validation before the function body executes and return 400 in case of validation failure. You can refer to its documentation [here](https://falcon.readthedocs.io/en/stable/api/media.html#validating-media).

`get_media` is a method defined on the Falcon `Request` object. You can refer to its documentation [here](https://falcon.readthedocs.io/en/stable/api/request_and_response_wsgi.html#falcon.Request.get_media).